templates/arm/deployment.json.erb

```diff
@@ -173,7 +173,24 @@
               "direction": "Inbound"
             }
           }
-          <% end %>
+        <% end %>
+        <% endpoints.each_with_index do |ports, index| %>
+          ,
+          {
+            "name": "custom_rule_<%= index %>",
+            "properties": {
+              "description": "Custom opened ports.",
+              "protocol": "Tcp",
+              "sourcePortRange": "*",
+              "destinationPortRange": "<%= ports %>",
+              "sourceAddressPrefix": "*",
+              "destinationAddressPrefix": "*",
+              "access": "Allow",
+              "priority": <%= 133 + index %>,
```

Comment:
Per the docs for network security group rule priority (https://docs.microsoft.com/en-us/azure/virtual-network/virtual-networks-nsg#Nsg-rules) rule priority must be between `Number between 100 and 4096`. I think this is an edge case, but might warrant an error if the user has `endpoints.length * 133 > 4096`.